### PR TITLE
docs: migration guide

### LINKED ISSUE #2112: Write a migration guide for V10
https://alpha.trpc.io/docs/migrate-from-v9-to-v10

Can be split into different parts

The main chunks of this can be found [here](https://github.com/trpc/trpc/blob/next/.tmp/v10-docs.md#interopability-mode-for-old-routers--migration-path): 

95% use-case:

- [x] Transforming old API for backward-compatabiltiy
- [ ] Add your first router in the new way
- [ ] Briefly explain new Router API
- [ ] Briefly ew client API with proxy
- [ ] [...]

The 5%:

- [ ] New Links
- [ ] New subscriptions with breaking changes
- [ ] [...]

### PR DESCRIPTION
**Pull request by juliusmarminge**
#2112

## 🎯 Changes

What changes are made in this PR? Is it a feature or a bug fix?

- Adds missing parts of the migration guide as well as some general improvements.

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/main/CONTRIBUTING.md).
- [x] If necessary, I have added documentation related to the changes made.
- [ ] I have added or updated the tests related to the changes made.

### CODE REVIEW

**Review by KATT — APPROVED** [review]
Looks like a great improvement!

File: www/docs/migration/migrate-from-v9-to-v10.mdx [hunk 1]
```diff
@@ -157,70 +175,82 @@ If you are migrating from V9->V10, the transition is pretty simple for the 95% u
...
+
+  /* [...] */
+
+- export const appRouter = t.mergeRouters(legacyRouter);
++ export const appRouter = t.mergeRouters(legacyRouter, greetingRouter);
 ```
 
+### Limitations of interopability mode
+
+Any procedures defined using `t.procedure` must be called on the client using the [new way of calling procedures](#calling-procedures) and vice-versa for old procedures. This means you have to move over an entire router on both the backend and frontend at once.
```
Comment (sachinraja) [inline www/docs/migration/migrate-from-v9-to-v10.mdx:252]: > This means you have to move over an entire router on both the backend and frontend at once.

I don't think this is true. You can have an interop router and a new router at the same time and migrate procedures over.

**Review by sachinraja — APPROVED** [review]
Love this!

File: www/docs/migration/migrate-from-v9-to-v10.mdx (hunk 1 above)
Comment (KATT) [reply www/docs/migration/migrate-from-v9-to-v10.mdx:252]: +1

**Comment by KATT** [comment]
Will merge this, it's still an improvement even if we can do more! thank you so much @juliusmarminge

File: www/docs/migration/migrate-from-v9-to-v10.mdx (hunk 1 above)
Comment (juliusmarminge) [reply www/docs/migration/migrate-from-v9-to-v10.mdx:252]: Ah yeah. Let me rewrite that.

What I meant what when you move that legacyRouter over to be a `t.router` you need to also move the usage of that router on the frontend

**Pull request merged**